jax/_src/lax/lax.py

```diff
@@ -4736,17 +4805,43 @@ def _reduce_number_dtype_rule(name, operand, *args, **kw):
                     "of number.".format(name, np.dtype(operand.dtype).name))
   return dtypes.canonicalize_dtype(operand.dtype)
 
-def _reduce_sum_shape_rule(operand, *, axes):
+def _reduce_sum_shape_rule(operand, *, axes, **kw):
   return _reduce_op_shape_rule(operand, axes=axes)
 
-def _reduce_sum_translation_rule(c, operand, *, axes):
-  dtype = c.get_shape(operand).numpy_dtype()
+def _reduce_sum_translation_rule(c, operand, *, axes, use_kahan_summation):
+  shape = c.get_shape(operand)
+  dtype = shape.numpy_dtype()
   scalar = ShapedArray((), dtype)
-  return xops.Reduce(c, [operand], [xb.constant(c, np.array(0, dtype))],
-                     xla.primitive_subcomputation(add_p, scalar, scalar),
-                     axes)
+  zero = xb.constant(c, np.array(0, dtype))
+
+  if not use_kahan_summation:
+    return xops.Reduce(c, [operand], [xb.constant(c, np.array(0, dtype))],
+                       xla.primitive_subcomputation(add_p, scalar, scalar),
+                       axes)
+
```

Comment:
Another trick you may prefer, but it's up to you (this is fine also):

Use `xla.lower_fun` and call your new `lax.reduce` inside the translation rule.

Reply:
I'm okay with the implementation as an XLA subcomputation; I think I'll leave as-is.